samples/CMakeLists.txt

```diff
@@ -33,6 +33,8 @@ if (CMAKE_SYSTEM_NAME MATCHES "Windows")
   add_definitions(-DNOMINMAX -DVK_USE_PLATFORM_WIN32_KHR)
 elseif(CMAKE_SYSTEM_NAME MATCHES "Linux")
   add_definitions(-DVK_USE_PLATFORM_XCB_KHR)
+elseif(CMAKE_SYSTEM_NAME MATCHES "Darwin")
+  add_definitions(-DVK_USE_PLATFORM_MACOS_MVK)
```

Comment:
@ylz-at this should actually prevent the error you're experiencing. Can you debug why the XCB path is taken? What is `CMAKE_SYSTEM_NAME` on your system?

Reply:
`vulkan.h:59:10 fatal error: 'X11/Xlib.h' file not found.` for which `.cpp` file you get this error message?

Reply:
@theHamsta CMAKE_SYSTEM_NAME is `Darwin`.

Reply:
My fault...`macos` branch can compile. I was using `master` branch just now. LoL

Reply:
Here is the output of cmake in case you are interested in,
```
"/Users/falco/Library/Application Support/JetBrains/Toolbox/apps/CLion/ch-0/221.5787.29/CLion.app/Contents/bin/cmake/mac/bin/cmake" -DCMAKE_BUILD_TYPE=Debug "-DCMAKE_MAKE_PROGRAM=/Users/falco/Library/Application Support/JetBrains/Toolbox/apps/CLion/ch-0/221.5787.29/CLion.app/Contents/bin/ninja/mac/ninja" -DSAMPLE_BUILD=ON -DSAMPLES_BUILD=ON -DSAMPLES_BUILD_ONLY_DYNAMIC=ON -DSAMPLES_BUILD_WITH_LOCAL_VULKAN_HPP=ON -DTESTS_BUILD=ON -DTESTS_BUILD_ONLY_DYNAMIC=ON -DTESTS_BUILD_WITH_LOCAL_VULKAN_HPP=ON -DVULKAN_HPP_RUN_GENERATOR=ON -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_CXX_STANDARD=11 -G Ninja -S /Users/falco/CLionProjects/Vulkan-Hpp -B /Users/falco/CLionProjects/Vulkan-Hpp/cmake-build-debug
-- The C compiler identification is AppleClang 13.1.6.13160021
-- The CXX compiler identification is AppleClang 13.1.6.13160021
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
--  Found clang-format version <14.0.5>.
--  Using .clang-format version 12.
CMAKE_CXX_STANDARD = <11>
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Using Cocoa for window creation
-- Found PythonInterp: /Library/Frameworks/Python.framework/Versions/3.10/bin/python3 (found suitable version "3.10", minimum required is "3") 
-- Google Mock was not found - tests based on that will not build
-- spirv-tools not linked - illegal SPIRV may be generated for HLSL
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    SAMPLE_BUILD


-- Build files have been written to: /Users/falco/CLionProjects/Vulkan-Hpp/cmake-build-debug

[Finished]
```